### PR TITLE
Add missed options for av1 encoder

### PR DESCRIPTION
**Pull request by xhaihao**

### CODE REVIEW

**Comment by xhaihao** [comment]
@wenbinc-Bin  could you have a look at this patch ?

**Comment by wenbinc-Bin** [comment]
LGTM.

**Review by dvrogozh — COMMENTED** [review]
I guess this needs to be squashed into av1 enabling patching considering that it is not yet merged into ffmpeg.

File: patches/0082-lavc-qsvenc_av1-add-missed-options.patch [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+ @section snow
+diff --git a/libavcodec/qsvenc.c b/libavcodec/qsvenc.c
+index 09777414f2..6de4811028 100644
+--- a/libavcodec/qsvenc.c
++++ b/libavcodec/qsvenc.c
+@@ -1012,6 +1012,8 @@ static int init_video_param(AVCodecContext *avctx, QSVEncContext *q)
+ 
+             q->extparam_internal[q->nb_extparam_internal++] = (mfxExtBuffer *)&q->extco2;
+         } else if (avctx->codec_id == AV_CODEC_ID_AV1) {
++            if (q->extbrc >= 0)
```
Comment (dvrogozh) [inline patches/0082-lavc-qsvenc_av1-add-missed-options.patch:38]: I wonder, do we need to start combining AV1 path with AVC and HEVC? would be more visible with the squashed single AV1 patch variant.

**Comment by xhaihao** [comment]
Yes, we'll squash the patches into the main av1 enc patch when submitting patchset to FFmpeg mailing list for review soon.

File: patches/0082-lavc-qsvenc_av1-add-missed-options.patch (hunk 1 above)
Comment (xhaihao) [reply patches/0082-lavc-qsvenc_av1-add-missed-options.patch:38]: So far there are lots of options are available for AVC and HEVC only, so we don't combine av1 path with avc and hevc path.

Comment (dvrogozh) [reply patches/0082-lavc-qsvenc_av1-add-missed-options.patch:38]: ok, was just reminding that this might be considered in a future.

Comment (xhaihao) [reply patches/0082-lavc-qsvenc_av1-add-missed-options.patch:38]: Yes, We'll combine them if possible in the future.

**Pull request merged**